integrations/pagerDuty.js

```diff
@@ -132,73 +139,81 @@ async function createIncident(authData, user) {
+async function searchIncidents(authData) {
   // Init our PD client
   const { pagerDutyKey } = authData
   pd.initializePagerDuty(pagerDutyKey)
 
   // Get query information from the user
-  const query = await ux.prompt(searchIncidentsPrompt)
+  const query = await getSearchQuery()
 
-  sdk.log('')
+  await ux.print('')
   ux.spinner.start('Retrieving incidents')
   const incidents = await pd.getIncidents(query)
   ux.spinner.stop('Done!\n')
 
   // Nothing found; return early
   if (!incidents.length) {
-    sdk.log(ux.colors.blue('No incidents found!'))
+    await ux.print(ux.colors.blue('No incidents found!'))
     return incidents
   }
 
-  sdk.log(ux.colors.magenta('\nWe found the following incidents:\n'))
-  incidents.map(printIncident)
+  await ux.print(ux.colors.magenta('\nWe found the following incidents:'))
+  const incidentsStr = incidents.map(printIncident)
+  await ux.print(incidentsStr.join(''))
 }
 
 /**
  * printIncident pretty prints an incident.
  */
 function printIncident(incident) {
-  sdk.log(`\t${incident.title}`)
-  sdk.log(`\t\tStatus: ${incident.status}`)
-  sdk.log(`\t\tService: ${incident.service.summary}`)
-  sdk.log(`\t\tEscalation Policy: ${incident.escalation_policy.summary}`)
-  sdk.log(`\t\tUrgency: ${incident.urgency}`)
-  sdk.log(`\t\tCreated at: ${incident.created_at}`)
-  sdk.log(`\t\tLast modified: ${incident.last_status_change_at}`)
+  let incidentStr = `\n\t${incident.title}
+  \t\tStatus: ${incident.status}
+  \t\tService: ${incident.service.summary}
+  \t\tEscalation Policy: ${incident.escalation_policy.summary}
+  \t\tUrgency: ${incident.urgency}
+  \t\tCreated at: ${incident.created_at}
+  \t\tLast modified: ${incident.last_status_change_at}`
   if (incident.assignments.length) {
-    sdk.log(`\t\tPeople assigned:`)
-    incident.assignments.map(assignee => {
-      sdk.log(`\t\t\t${asignee.summary}`)
+    incidentStr += `\n\t\tPeople assigned:`
+    incident.assignments.map(async assignee => {
+      incidentStr += `\n\t\t\t${assignee.summary}`
     })
   }
-  sdk.log(`\t\tLink: ${incident.html_url}`)
+  incidentStr += `\n\t\tLink: ${incident.html_url}\n`
+  return incidentStr
 }
 
 /**
  * whoOnCall retrieves users on call with PagerDuty and prints them out
  *
  * @param {object} authData The user's entered auth data
- * @param {object} user     The current SDK user
  */
-async function whoOnCall(authData, user) {
+async function whoOnCall(authData) {
   // Init our PD client
   const { pagerDutyKey } = authData
   pd.initializePagerDuty(pagerDutyKey)
+  const { callOutCyan } = ux.colors
 
   // Pretty print the on call users for each policy
+  // Concatenated strings before printing for ux.print
   const onCalls = await pd.getOnCall()
-  Object.keys(onCalls).map(policy => {
-    sdk.log(
-      ux.colors.magenta(
-        `The following people are on call for the '${policy}' policy:\n\n`
-      )
+  Object.keys(onCalls).forEach(async policy => {
+    const policyTitle = ux.colors.magenta(
```

Comment:
you deconstructed callOutCyan right above, would you mind cleaning up this codebase by pulling out magenta and blue everywhere too? The pattern with the ux.colors. boilerplate is a bit of a pet peeve lol

Reply:
Also pull out all the colors at the top of the file <3

Reply:
done!